xen/arch/arm/coproc/coproc.c

```diff
@@ -96,6 +119,92 @@ static struct coproc_device *coproc_find_by_path(const char *path)
     return found ? coproc : NULL;
 }
 
+static struct vcoproc_instance *coproc_init_vcoproc(struct domain *d,
+                                                    struct coproc_device *coproc)
+{
+    struct vcoproc_instance *vcoproc;
+    int ret;
+
+    vcoproc = xzalloc(struct vcoproc_instance);
+    if ( !vcoproc )
+    {
+        printk("Failed to allocate vcoproc_instance for %s\n",
+               dev_path(coproc->dev));
+        return ERR_PTR(-ENOMEM);
+    }
+
+    vcoproc->coproc = coproc;
+    vcoproc->domain = d;
+    vcoproc->state = VCOPROC_UNKNOWN;
+    spin_lock_init(&vcoproc->lock);
+
+    ret = coproc->ops->vcoproc_init(d, coproc, vcoproc);
```

Comment:
Why do you need a coproc here? You already put it into the vcoproc few lines above.